day003/src/test/java/com/pawelpluta/day003/ParameterizedTest.java

```diff
@@ -11,47 +13,36 @@
 
 class ParameterizedTest {
 
-    private DummyCalculator calculator = new DummyCalculator();
+    private final DummyCalculator calculator = new DummyCalculator();
 
-    @Test
-    void twoPositiveNumbersCanBeAdded() {
-        // expect
-        assertEquals(valueOf(5), calculator.sum(valueOf(3), valueOf(2)));
-    }
 
-    @Test
-    void negativeAndPositiveNumbersCanBeAdded() {
-        // expect
-        assertEquals(valueOf(-1), calculator.sum(valueOf(-3), valueOf(2)));
+    private static Stream<Arguments> dataForCorrectSumCalculation() {
+        return Stream.of(
+                Arguments.of("two positive numbers can be added", 3, 2, 5),
+                Arguments.of("negative and positive numbers can be added", -3, 2, -1),
+                Arguments.of("two negative numbers can be added", -3, -2, -5)
+        );
     }
 
-    @Test
-    void twoNegativeNumbersCanBeAdded() {
-        // expect
-        assertEquals(valueOf(-5), calculator.sum(valueOf(-3), valueOf(-2)));
+    private static Stream<Arguments> dataForException() {
+        return Stream.of(
+                Arguments.of("null as first number is unsupported", null, ONE),
+                Arguments.of("null as second number is unsupported", ONE, null),
+                Arguments.of("null as both numbers is unsupported", null, null)
+        );
     }
 
-    @Test
-    void nullAsFirstNumberIsUnsupported() {
-        // expect
-        assertThrows(
-                IllegalArgumentException.class,
-                () -> calculator.sum(null, ONE));
-    }
-
-    @Test
-    void nullAsSecondNumberIsUnsupported() {
-        // expect
-        assertThrows(
-                IllegalArgumentException.class,
-                () -> calculator.sum(ONE, null));
+    @org.junit.jupiter.params.ParameterizedTest(name = "{index}. {0}")
+    @MethodSource("dataForCorrectSumCalculation")
```

Comment:
Ciekawostka - jeżeli nazwa metody testowej i metody tworzącej dane są takie same, to nie ma potrzeby przekazywania parametry w adnotacji.